### DISCUSSION TITLE
asix-ax88179 failed to build on macOS 12.1

### DISCUSSION THREAD

**Discussion by h31** [Homebrew/discussions#2871] [Casks]
Hello! Looks like a formula problem, however I'm not sure that it is reproducible on other machines.
```
artyom@Artyom-MacBook-Pro ~ [1]> brew doctor
Your system is ready to brew.
```

```
artyom@Artyom-MacBook-Pro ~> brew upgrade asix-ax88179
==> Upgrading 1 outdated package:
asix-ax88179 2.0.0,1179 -> 2.0.0,1179
==> Upgrading asix-ax88179
==> Caveats
You must reboot for the installation of asix-ax88179 to take effect.

==> Downloading https://www.asix.com.tw/en/support/download/file/1179
Already downloaded: /Users/artyom/Library/Caches/Homebrew/downloads/ec30988aaa90da54456ca60b82db6ffbe5edf2342fb3da6cb92453674d55ff6a--ASIX_USB_Device_Installer_macOS_12_Driver_v2.0.0.zip
Warning: No checksum defined for cask 'asix-ax88179', skipping verification.
==> Purging files for version 2.0.0,1179 of Cask asix-ax88179
Error: asix-ax88179: undefined method `rename' for nil:NilClass
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/cask/cmd/upgrade.rb:160:in `upgrade_casks'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:238:in `upgrade_outdated_casks'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:111:in `upgrade'
/usr/local/Homebrew/Library/Homebrew/brew.rb:110:in `<main>'
```

```
artyom@Artyom-MacBook-Pro ~ [1]> brew config
HOMEBREW_VERSION: 3.3.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 768b172393339b4654107dc6a840a2cd08dc64b9
Last commit: 7 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a0e81be5dbf1706ab0c9e4f482df91e726f3027f
Core tap last commit: 5 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 13.0.0 build 1300
Git: 2.34.1 => /usr/local/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.1-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
```

**Comment by carlocab** [comment]
What's the _full_ output of
```
brew doctor --verbose
brew upgrade --verbose --debug asix-ax88179
```
?

**Reply by h31** [reply]
[brew doctor --verbose](https://gist.github.com/h31/5bb53bd2d65d2d7f48bfa8cb6c013716)
[brew upgrade --verbose --debug asix-ax88179](https://gist.github.com/h31/14f7cc7b814a7242e40672c07d5457c6)

**Reply by carlocab** [reply]
Hmm. That's weird. You're already supposed to be on the latest version: `asix-ax88179 2.0.0,1179 -> 2.0.0,1179`

I'm not really sure what's going on here, I'm afraid. Please file an issue at [Homebrew/brew](https://github.com/Homebrew/brew.git).

**Reply by h31** [reply]
[Done](https://github.com/Homebrew/brew/issues/12802), thank you for the investigation!

**Reply by cho-m** [reply]
Not sure why message looks like that, but I see some logs showing path `/usr/local/Caskroom/asix-ax88179/.metadata/1.3.0,1163/`. I would guess you are going `1.3.0,1163 -> 2.0.0,1179`.

One potential reason could be if you upgraded from macOS Big Sur to Monterey since the above versions match the conditional sections in Cask https://github.com/Homebrew/homebrew-cask-drivers/blob/master/Casks/asix-ax88179.rb

I had previously triaged a similar issue for `asix-ax88179` at Homebrew/homebrew-cask-drivers#2297. This required some local/manual changes to successfully uninstall.

**Reply by h31** [reply]
Yes, I've upgraded from Big Sur. What should I do to uninstall the formula?

**Reply by gromgit** [reply]
Try the following:
```
brew uninstall asix-ax88179
rm -fr /usr/local/Caskroom/asix-ax88179
brew install asix-ax88179
```

**Reply by h31** [reply]
The first command (`brew uninstall asix-ax88179`) prints the same error: `Error: undefined method 'rename' for nil:NilClass`. Ever `--force` doesn't help.

**Reply by gromgit** [reply]
Sorry, forgot to mention the error is expected for the first command, but _not_ for the last. You need to do all three to address your situation.
